pkg/reconciler/knativeserving/knativeserving.go

```diff
@@ -124,35 +109,13 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, original *servingv1alpha1
 
 // ReconcileKind compares the actual state with the desired, and attempts to
 // converge the two.
-func (r *Reconciler) ReconcileKind(ctx context.Context, original *servingv1alpha1.KnativeServing) pkgreconciler.Event {
-	logger := logging.FromContext(ctx)
-
-	// Convert the namespace/name string into a distinct namespace and name
-	key, err := cache.MetaNamespaceKeyFunc(original)
-	if err != nil {
-		logger.Errorf("invalid resource key: %s", key)
-		return nil
-	}
-	r.servings[key] = original.Generation
-
-	// Reconcile this copy of the KnativeServing resource and then write back any status
-	// updates regardless of whether the reconciliation errored out.
-	err = r.reconcile(ctx, original)
-	if err != nil {
-		return err
-	}
-	return nil
-}
-
-func (r *Reconciler) reconcile(ctx context.Context, ks *servingv1alpha1.KnativeServing) error {
+func (r *Reconciler) ReconcileKind(ctx context.Context, ks *servingv1alpha1.KnativeServing) pkgreconciler.Event {
 	logger := logging.FromContext(ctx)
+	ks.Status.InitializeConditions()
```

Comment:
You sure we want to do this all the time? It's not going to overwrite anything important?

Reply:
Nope, serving's controllers do the same thing.